hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -403,7 +403,9 @@ protected void preWrite(String instantTime, WriteOperationType writeOperationTyp
         .isPresent()
         ? Option.of(lastCompletedTxnAndMetadata.get().getLeft()) : Option.empty());
     try {
-      syncTableMetadata();
+      if (writeOperationType != WriteOperationType.CLUSTER && writeOperationType != WriteOperationType.COMPACT) {
```

Comment:
What about AsyncClean? Should't we instead reverse the check:

if (writeOperationType == COMMIT or writeOperationType == DELTACOMMIT) {
 ...}

Reply:
Clean does not go thru preWrite/postCommit IIRC. So we should already be ok. I ll double check this and also callers of commitStats

Reply:
I ll reverse the check, to be conservative. Good callout

Reply:
So, the check is cumbersome to write otherwise. `writeOperationType` is upsert, delete, insert etc. not the action type. Will keep as-is

Reply:
Ok. fine this way too.